pkg/backup/actions/csi/pvc_action.go

```diff
@@ -570,3 +581,485 @@ func NewPvcBackupItemAction(f client.Factory) plugincommon.HandlerInitializer {
+	if err := p.crClient.List(ctx, vsList, crclient.InNamespace(ns)); err != nil {
+		return errors.Wrapf(err, "failed to list VolumeSnapshots in namespace %s", ns)
+	}
+
+	for i := range vsList.Items {
+		vs := vsList.Items[i]
+		if vs.Status == nil || vs.Status.VolumeGroupSnapshotName == nil || *vs.Status.VolumeGroupSnapshotName != vgs.Name {
+			continue
+		}
+
+		err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			// Re-fetch the latest VS to avoid conflict
+			latestVS := &snapshotv1api.VolumeSnapshot{}
+			if err := p.crClient.Get(ctx, crclient.ObjectKeyFromObject(&vs), latestVS); err != nil {
+				return errors.Wrapf(err, "failed to get latest VolumeSnapshot %s", vs.Name)
+			}
+
+			// Remove VGS owner ref
+			if err := controllerutil.RemoveOwnerReference(vgs, latestVS, p.crClient.Scheme()); err != nil {
+				return errors.Wrapf(err, "failed to remove VGS owner reference from VS %s", vs.Name)
+			}
+
+			// Remove known finalizers
+			controllerutil.RemoveFinalizer(latestVS, VolumeSnapshotFinalizerGroupProtection)
+			controllerutil.RemoveFinalizer(latestVS, VolumeSnapshotFinalizerSourceProtection)
+
+			// Add Velero labels
+			if latestVS.Labels == nil {
+				latestVS.Labels = make(map[string]string)
+			}
+			latestVS.Labels[velerov1api.BackupNameLabel] = backup.Name
+			latestVS.Labels[velerov1api.BackupUIDLabel] = string(backup.UID)
+
+			// Attempt to update
+			return p.crClient.Update(ctx, latestVS)
+		})
+		if err != nil {
+			return errors.Wrapf(err, "failed to update VS %s after retrying on conflict", vs.Name)
+		}
+	}
+
+	return nil
+}
+
+func (p *pvcBackupItemAction) patchVGSCDeletionPolicy(vgs *volumegroupsnapshotv1beta1.VolumeGroupSnapshot) error {
+	ctx := context.TODO()
+
+	vgscName := vgs.Status.BoundVolumeGroupSnapshotContentName
+	if vgscName == nil {
+		return errors.New("VolumeGroupSnapshotContent name not found in VGS status")
+	}
+
+	vgsc := &volumegroupsnapshotv1beta1.VolumeGroupSnapshotContent{}
+	if err := p.crClient.Get(ctx, crclient.ObjectKey{Name: *vgscName}, vgsc); err != nil {
+		return errors.Wrapf(err, "failed to get VolumeGroupSnapshotContent %s for VolumeGroupSnapshot %s/%s", *vgscName, vgs.Namespace, vgs.Name)
+	}
+
+	if vgsc.Spec.DeletionPolicy == snapshotv1api.VolumeSnapshotContentDelete {
+		p.log.Infof("Patching VGSC %s to Retain deletionPolicy", *vgscName)
+		vgsc.Spec.DeletionPolicy = snapshotv1api.VolumeSnapshotContentRetain
+		if err := p.crClient.Update(ctx, vgsc); err != nil {
+			return errors.Wrapf(err, "failed to update VGSC %s deletionPolicy", *vgscName)
+		}
+	} else {
+		p.log.Infof("VGSC %s already set to deletionPolicy=%s", *vgscName, vgsc.Spec.DeletionPolicy)
+	}
+
+	return nil
+}
+
+func (p *pvcBackupItemAction) deleteVGSAndVGSC(vgs *volumegroupsnapshotv1beta1.VolumeGroupSnapshot) error {
+	ctx := context.TODO()
+
+	vgscName := vgs.Status.BoundVolumeGroupSnapshotContentName
+	if vgscName != nil {
+		vgsc := &volumegroupsnapshotv1beta1.VolumeGroupSnapshotContent{}
```

Comment:
Just call deletion API should be good enough.

Reply:
Yes, call the delete API and log any error as wraning.